### ISSUE TITLE
ObjectStatus: Accessibility issues with reading ObjectStatus stateAnnouncementText and missing tooltips on icon in VoiceOver on Mac OS.

### ISSUE THREAD

**Issue by samanrasool** [SAP/ui5-webcomponents-react#5625]
### Describe the bug

The MacOs VoiceOver App does not read the `stateAnnouncementText` in the ObjectStatus component correctly. This issue has been identified in two ways:

1. When using the attributes `showDefaultIcon` and `state="Success"`, the screen reader should read "Success". However, this is not happening.

2. When only an icon, either the default icon or the Icon component, is used, hovering over the ObjectStatus should display a tooltip describing the icon. However, no tooltip is being provided.

Additionally, instead of reading the relevant information, VoiceOver is reading "Object Status" instead.

This is a bug and needs to be addressed as it does not meet the accessibility requirements for the application.

### Isolated Example

https://stackblitz.com/edit/github-8ksjkm?file=src%2FApp.tsx

### Reproduction steps

1. Open VoiceOver app on your mac book
2. Focus on Object status and notice what voiceover reads.

### Expected Behaviour

1. The screen reader should read state of the object status accordingly. 
2. Hovering over the object status containing only icon should display a tooltip

### Screenshots or Videos

![Screenshot 2024-03-19 at 9 05 15 PM](https://github.com/SAP/ui5-webcomponents-react/assets/160647561/46ebd4f9-8a10-46ab-904f-9ccc49b2198e)

### UI5 Web Components for React Version

1.26.0

### UI5 Web Components Version

1.23.0

### Browser

Chrome

### Operating System

Mac Os Sonoma 14.3.1

### Organization

SAP

### Declaration

- [X] I’m not disclosing any internal or sensitive information.

**Comment by Lukas742** [comment]
Hi @samanrasool 

the `ObjectStatus` should always include a text, only the icon is optional. (see [here](https://experience.sap.com/fiori-design-web/object-display-elements/#object-status)) We didn't made `children` mandatory as it's possible that the ObjectStatus is loading its content in runtime and therefore could initially be empty. 

We still decided to also announce the state in icon-only mode (see linked PR).

> When only an icon, either the default icon or the Icon component, is used, hovering over the ObjectStatus should display a tooltip describing the icon. However, no tooltip is being provided.

Since we don't support icon-only mode, this has to be implemented on your end by setting the `title` prop. Also, it's not recommended setting tooltips on non interactive elements, it's better to use a label for this. (See [here](https://experience.sap.com/fiori-design-web/using-tooltips/#usage))

> Additionally, instead of reading the relevant information, VoiceOver is reading "Object Status" instead.

This is intentional, visually impaired persons need to know that it's not just a text, but an ObjectStatus.
